### PR TITLE
golang/volume

### PR DESCRIPTION
**Pull request by ronaudinho**
example of volume usage, related to https://github.com/nanovms/ops/pull/598

### CODE REVIEW

File: golang/05-empty-volume/volume.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	for i := 0; i < 2; i++ {
+		readdir("mnt", i)
+		// on 1st try, non-existing, should err
+		err = readtxt(mounted("hello.txt"))
+		if err != nil {
+			fmt.Println(err)
+			continue
+		}
+		// write file
+		err = writetxt(mounted("hello.txt"), "lol")
```
Comment (ronaudinho) [inline golang/05-empty-volume/volume.go:29]: initially `hello.txt` contains the text `hello`, but replacing it with another text via `ioutil.WrifeFile` does not completely replace the text if the initial content is longer e.g expected new text to be `lol` but got `lollo` instead.

File: golang/05-empty-volume/volume.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
+package main
+
+import (
+	"encoding/json"
```
Comment (eyberg) [inline golang/05-empty-volume/volume.go:4]: can't compile w/this

File: golang/05-empty-volume/volume.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	"fmt"
+	"io/ioutil"
+	"log"
```
Comment (eyberg) [inline golang/05-empty-volume/volume.go:7]: can't compile w/this

File: golang/05-empty-volume/volume.go [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+func main() {
+	var err error
+	// read-write files
+	for i := 0; i < 2; i++ {
+		readdir("mnt", i)
+		// on 1st try, non-existing, should err
+		err = readtxt(mounted("hello.txt"))
+		if err != nil {
+			fmt.Println(err)
+			continue
```
Comment (eyberg) [inline golang/05-empty-volume/volume.go:26]: why are these here? it skips the next block?

File: golang/05-empty-volume/volume.go [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+		}
+		// write file
+		err = writetxt(mounted("hello.txt"), "lol")
+		if err != nil {
+			fmt.Println(err)
+			continue
```
Comment (eyberg) [inline golang/05-empty-volume/volume.go:32]: ??

File: golang/05-empty-volume/volume.go (hunk 1 above)
Comment (eyberg) [reply golang/05-empty-volume/volume.go:29]: i'm unable to reproduce...

File: golang/05-empty-volume/volume.go [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+var mounted = func(file string) string { return path.Join("mnt", file) }
+
+type hello struct {
+	Message string `json:"message"`
+}
+
+func main() {
+	var err error
+	// read-write files
```
Comment (eyberg) [inline golang/05-empty-volume/volume.go:19]: you are missing a os.Mkdir somewhere around here unless the dirs are being created by the mount?

File: golang/05-empty-volume/volume.go [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"path"
+)
+
+var mounted = func(file string) string { return path.Join("mnt", file) }
```
Comment (eyberg) [inline golang/05-empty-volume/volume.go:11]: /mnt

File: golang/05-empty-volume/volume.go [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+type hello struct {
+	Message string `json:"message"`
+}
+
+func main() {
+	var err error
+	// read-write files
+	for i := 0; i < 2; i++ {
+		readdir("mnt", i)
```
Comment (eyberg) [inline golang/05-empty-volume/volume.go:21]: /mnt

File: golang/06-non-empty-volume/volume.go [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path"
+)
+
+var mounted = func(file string) string { return path.Join("mnt", file) }
```
Comment (eyberg) [inline golang/06-non-empty-volume/volume.go:11]: /mnt

File: golang/06-non-empty-volume/volume.go [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+
+type hello struct {
+	Message string `json:"message"`
+}
+
+func main() {
+	var err error
+	// read-write files
+	for i := 0; i < 2; i++ {
+		readdir("mnt", i)
```
Comment (eyberg) [inline golang/06-non-empty-volume/volume.go:21]: /mnt

File: golang/05-empty-volume/volume.go (hunk 2 above)
Comment (ronaudinho) [reply golang/05-empty-volume/volume.go]: fixed

File: golang/05-empty-volume/volume.go (hunk 3 above)
Comment (ronaudinho) [reply golang/05-empty-volume/volume.go]: fixed

File: golang/05-empty-volume/volume.go (hunk 6 above)
Comment (ronaudinho) [reply golang/05-empty-volume/volume.go:17]: dirs are created by the mount, which might be an issue since users will have to prefix the expected mount path to their file related functions, or am i getting this wrong?

Comment (eyberg) [reply golang/05-empty-volume/volume.go:17]: got it; relative mounts are fine as long as the user is aware; i just retried the example && that works; thanks!
